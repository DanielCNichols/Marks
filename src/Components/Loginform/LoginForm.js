import React, { useState } from 'react';
import { useLoginForm, validationRules } from '../../Hooks/useLoginForm';
import AuthApiService from '../../Services/authService';
import TokenService from '../../Services/tokenService';

export default function LoginForm(props) {
  const [error, setError] = useState(null);
  const { inputs, handleChange, handleSubmit, inputErrors } = useLoginForm(
    handleLogin,
    validationRules
  );

  async function handleLogin(inputs) {
    try {
      let { token } = await AuthApiService.postLogin(inputs);
      TokenService.saveAuthToken(token);
      props.history.push('/list');
    } catch ({ error }) {
      setError(error);
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <fieldset>
        <legend>Login</legend>
        <div className="form-element">
          <label htmlFor="username">Username</label>
          <input
            type="text"
            name="username"
            id="username"
            onChange={handleChange}
            value={inputs.username}
          />
        </div>
        <div className="form-element">
          <label htmlFor="password">Password</label>
          <input
            type="text"
            name="password"
            id="password"
            onChange={handleChange}
            value={inputs.password}
          />
        </div>

        {inputErrors && (
          <>
            {Object.keys(inputErrors).map((e, idx) => {
              return (
                <p style={{ color: 'white' }} key={idx}>
                  {inputErrors[e]}
                </p>
              );
            })}
          </>
        )}

        {error && <p className="error">{error}</p>}

        <button type="submit">Login</button>
      </fieldset>
    </form>
  );
}
