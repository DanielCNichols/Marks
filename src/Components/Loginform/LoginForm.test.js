import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import LoginForm from './LoginForm';
import { BrowserRouter } from 'react-router-dom';

describe('LoginForm tests', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <LoginForm />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <LoginForm />
        </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
