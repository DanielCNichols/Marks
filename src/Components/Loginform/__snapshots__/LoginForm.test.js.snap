// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm tests renders the UI as expected 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <fieldset>
    <legend>
       Sign In
    </legend>
    <div
      className="formElement"
    >
      <label
        htmlFor="username"
      >
        Username
      </label>
      <input
        aria-placeholder="daniel61"
        aria-required="true"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="daniel61"
        type="text"
        value=""
      />
    </div>
    <div
      className="formElement"
    >
      <label
        htmlFor="password"
      >
        Password
      </label>
      <input
        aria-placeholder="Pass123"
        aria-required="true"
        autoComplete="password"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="Pass123"
        type="password"
        value=""
      />
    </div>
    <button
      className="submit"
      type="submit"
    >
      Login
    </button>
  </fieldset>
</form>
`;
